### PR TITLE
Beefy pallet test

### PR DESCRIPTION
**Pull request by adoerr**
This PR provides a basic pallet test infrastructure before migrating the Beefy pallet to FRAMEv2.

### CODE REVIEW

File: beefy-pallet/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg(test)]
```
Comment (tomusdrw) [inline beefy-pallet/src/tests.rs:17]:
```suggestion
```
likewise

File: beefy-pallet/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		assert_eq!(want[1], authorities[1]);
+	});
+}
+
+#[test]
+fn session_change_updates_authorities() {
+	new_test_ext(vec![1, 2, 3, 4]).execute_with(|| {
+		init_block(1);
+
+		assert!(0 == Beefy::validator_set_id());
```
Comment (tomusdrw) [inline beefy-pallet/src/tests.rs:65]: Might be worth checking `Beefy::authorities()` and `Beefy::next_authorities()` here. I think the next storage entry should be updated after genesis.

**Review by tomusdrw — COMMENTED** [review]
lgtm! couple of minor things

File: beefy-pallet/Cargo.toml [hunk 3]
```diff
@@ -13,10 +13,17 @@ beefy-primitives = { path = "../primitives", default-features = false }
...
 frame-support = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
 frame-system = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
+
 sp-runtime = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
 sp-std = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
+
 pallet-session = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "master" }
 
+[dev-dependencies]
+sp-core = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
```
[the remark this answers is not in the document]
Comment (adoerr) [reply beefy-pallet/Cargo.toml:23]: I don;t think so, as long as `sp-core` is not used in parallel as a dependency in the same crate. Besides, it seems to be common practice. I checked with a couple of other pallets.

File: beefy-pallet/src/tests.rs (hunk 2 above)
Comment (adoerr) [reply beefy-pallet/src/tests.rs:64]: I have added a separate test for `Beefy::next_authorities()` update. The way `OneSessionHandler` is currently implemented, next authorities *only* update if authorities themselves update as well. In other words, for session `0` and `1` both, authorities and next authorities don't update. Let's discuss this during the upcoming sync-up.

**Review by tomusdrw — APPROVED** [review]
lgtm!

**Pull request merged**